pennylane/utils.py

```diff
@@ -353,3 +356,30 @@ def __str__(self):
                 output += "{}({}(wires={}))\n".format(return_map[op.return_type], op.name, op.wires)
 
         return output
+
+
+def inv(operation_list):
+    """Invert a list of operations or a template.
+
+    If the inversion happens inside a QNode, the operations are removed and requeued
+    in the reversed order for proper inversion.
+
+    Args:
+        operation_list (Sequence[~.Operation]): A sequence of operations
+
+    Returns:
+        Sequence[~.Operation]: The inverted sequence of operations
+    """
+    inv_ops = [op.inv() for op in reversed(copy.deepcopy(operation_list))]
+
+    if qml._current_context is not None:
+        indices = [qml._current_context.queue.index(op) for op in operation_list]
```

Comment:
This line might need some modification, as it might be, that each time ``qml._current_context.queue.index(op)`` would return the very first operation it comes across (of the same operation). This means, that it might be, that there are several ``RX`` rotations in the ``operation_list``, and for each we would get the same index (see the failing example test I included later).

Reply:
Maybe getting the ``id`` of the operation could help (provided that exactly the same objects are passed)?

Reply:
Same as below:

> Will not be an issue. The operations are always uniquely identified via their ID:
> 
> ```python
> >>> import pennylane as qml
> >>> qml.PauliX(0) == qml.PauliX(0)
> False
> ```